rocAL/rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -65,97 +60,96 @@ def __next__(self):
             print("Process  time ::",timing_info.process_time)
             print("Transfer time ::",timing_info.transfer_time)
             raise StopIteration
-
-        if self.loader.run() != 0:
+        if self.loader.rocalRun() != 0:
             raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.rocalGetOutputTensors()
 
-        if(types.NCHW == self.tensor_format):
-            self.loader.copyToTensorNCHW(self.out, self.multiplier, self.offset, self.reverse_channels, int(self.tensor_dtype))
+        self.augmentation_count = len(self.output_tensor_list)
+        self.w = self.output_tensor_list[0].batch_width()
+        self.h = self.output_tensor_list[0].batch_height()
+        self.batch_size = self.output_tensor_list[0].batch_size()
+        self.color_format = self.output_tensor_list[0].color_format()
+        if self.tensor_dtype == types.FLOAT:
+            data_type="float32"
+        elif self.tensor_dtype == types.FLOAT16:
+            data_type="float16"
         else:
-            self.loader.copyToTensorNHWC(self.out, self.multiplier, self.offset, self.reverse_channels, int(self.tensor_dtype))
+            data_type= "uint8"
+        self.out = np.zeros(( self.batch_size * self.augmentation_count, self.h, self.w, self.color_format),dtype=data_type)
         
         if(self.loader._name == "TFRecordReaderDetection"):
             self.bbox_list =[]
             self.label_list=[]
             self.num_bboxes_list=[]
-            #Count of labels/ bboxes in a batch
-            self.bboxes_label_count = np.zeros(self.bs, dtype="int32")
-            self.count_batch = self.loader.GetBoundingBoxCount(self.bboxes_label_count)
-            self.num_bboxes_list = self.bboxes_label_count.tolist()
-            # 1D labels array in a batch
-            self.labels = np.zeros(self.count_batch, dtype="int32")
-            self.loader.GetBBLabels(self.labels)
-            # 1D bboxes array in a batch
-            self.bboxes = np.zeros((self.count_batch*4), dtype="float32")
-            self.loader.GetBBCords(self.bboxes)
-            #1D Image sizes array of image in a batch
-            self.img_size = np.zeros((self.bs * 2),dtype = "int32")
+            if(data_type== "uint8"):
+                self.output_tensor_list[0].copy_data_numpy(self.out)
+            else: 
```

Comment:
Now that we have same names for the functions which would work for uint8 & fp16/fp32 dtypes, why is the if else condition required ?
Please remove this